src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/commands.py

```diff
@@ -7,24 +7,44 @@
 from azure.cli.command_modules.consumption._transformers import transform_usage_list_output
 from azure.cli.command_modules.consumption._transformers import transform_reservation_summaries_list_output
 from azure.cli.command_modules.consumption._transformers import transform_reservation_details_list_output
+from azure.cli.command_modules.consumption._transformers import transform_pricesheet_get_output
 from azure.cli.command_modules.consumption._client_factory import usage_details_mgmt_client_factory
 from azure.cli.command_modules.consumption._client_factory import reservations_summaries_mgmt_client_factory
 from azure.cli.command_modules.consumption._client_factory import reservations_details_mgmt_client_factory
+from azure.cli.command_modules.consumption._client_factory import pricesheet_mgmt_client_factory
 from ._exception_handler import consumption_exception_handler
 from ._validators import validate_both_start_end_dates
 from ._validators import validate_reservations_summaries
 from ._validators import validate_reservations_details
+from ._validators import validate_pricesheet
+from ._validators import validate_usage_bp_inputs
 
 
 def load_command_table(self, _):
     with self.command_group('consumption usage') as g:
         g.custom_command('list', 'cli_consumption_list_usage', transform=transform_usage_list_output,
                          exception_handler=consumption_exception_handler, validator=validate_both_start_end_dates, client_factory=usage_details_mgmt_client_factory)
 
+        g.custom_command('billing period list', 'cli_consumption_list_usage_by_billing_period', transform=transform_usage_list_output,
```

Comment:
This looks like it creates an subgroup `billing` under `usage` that then only contains a single subgroup `period` which then contains only a single command `list`.  There should be commands under billing, and more commands under period.  If that's not the case, the command name should be changed. We should discuss.

Reply:
@tjprescott  intent is that - under the 'consumption usage' group there, there are 2 subgroups, 'list' and 'list by billing period', given the best practices doc, the following should work?
subgroups: 'list' and 'billing-period-list'. 
Please confirm

Reply:
@tjprescott  List by Billing Period expects the user to specify BillingPeriodId and the result only has items associated with that Id. Unlike 'list', list by billing period, marks billingperiodid as a required parameter. I'm not sure how we'd fold the two into a single command

Reply:
Why could you not call one command vs the other based on whether BillingPeriodId is specified?

Reply:
```Python
def list(client, billing_period=None):
  if billing_period:
    return client.list_by_billing_period(billing_period)
  return client.list()
```